src/middlewares/accomodation.middleware.js

```diff
@@ -39,5 +39,31 @@ class AccomodationMiddleware {
     }
     return next();
   }
+
+  /** checks if the accomodation selected is available
+   *  @param {req} req it contains the request of the user
+   *  @param {res} res it contains the response the user receive
+   *  @param {function} next it jumps to the next middleware in the route
+   * @return {object} the data from the first middleware
+   */
+  static async checkBookingRequestAvailability(req, res, next) {
+    const departureDate = new Date(req.body.departureDate);
+    const checkoutDate = new Date(req.body.checkoutDate);
+    if (checkoutDate <= departureDate) {
+      return response.errorMessage(res, 'The departure date cannot be later than the return date', 400);
+    }
```

Comment:
@VedaDom Here you're doing some validation rather than checking if the accommodation or room is available. This should be moved to another function/middleware that caters for validation